cocos/audio/android/AudioDecoder.cpp

```diff
@@ -468,8 +477,11 @@ bool AudioDecoder::decodeToPcm()
 
     ALOGV("Stopped decoding");
 
-    /* Destroy the UrlAudioPlayer object */
-    SL_DESTROY_OBJ(_playObj);
+    /* Destroy the AudioPlayer object */
+    {
+        std::lock_guard<std::mutex> lk(__SLPlayerMutex);
+        SL_DESTROY_OBJ(_playObj);
+    }
```

Comment:
Could you remove `}` after SL_DESTROY_OBJ, but keep it after `->Realize`, and test it again?